### ISSUE TITLE
Handle accidental `git lfs [smudge | clean]` runs

### ISSUE THREAD

**Issue by technoweenie** [github/git-lfs#210]
These are plumbing commands that aren't designed to be run by users.  However, running them should not hang indefinitely for users.

**Comment by brabil** [comment]
I was experimenting with Git LFS and I experienced this hanging on git lfs clean/smudge. I was puzzled on what is the recommended way/commands for end users to fetch their binaries from LFS Server without these plumbing  [smudge|clean] commands ?

**Comment by technoweenie** [comment]
I actually had a PR for a `git lfs download` command, but it never saw any (internal) comments, and is now way out of date.  But I do plan to add it soon.

Here's an ugly workaround though:

First, get the blob OID containing the text pointer:

```
$ git ls-tree master -rl | grep droidtocat
100644 blob 50eb175bbfde63504add621427a6a98910e89697     130	droidtocat.gif
100644 blob c6311098f7d0d260058c01fc69c5c5b68e24220b     131	droidtocat.jpg
```

Then you can print out the blob's pointer:

```
$ git cat-file -p 50eb175bbfde63504add621427a6a98910e89697
version https://git-lfs.github.com/spec/v1
oid sha256:d1c8fab51418ef587fcf5dd8e73c60b9700704e2f8f5292ea12ec27c285b23a3
size 25237
```

Just pipe that to `git lfs smudge`:

```
# yes, you have to direct the output to the correct file name :(
$ git cat-file -p 50eb175bbfde63504add621427a6a98910e89697 | git lfs smudge > droidtocat.gif
```

If you're interested, I'll ping you on the PR that adds the download command.

**Comment by evlacan** [comment]
Looks like a promising approach. There is still a small issue, the file ends up in two places  at this point: as  droidtocat.gif and as 50eb175bbfde63504add621427a6a98910e89697 file and under .git/lfs/objects/... Which looks like redundant at this point and takes double disk space. Do you have plans to do the clean-up as well?

**Comment by technoweenie** [comment]
The files in `.git/lfs/objects` act as a local cache, so clients don't have to re-download files as they change branches or checkout older commits.  We do want some kind of `git lfs gc` command to clean it up, but we're not totally sure how it'll work.

Do you need it to be ruthless in clearing up disk space, or would you be comfortable with it keeping the last 30 days or so of files?  It's likely there'll be various knobs you can tune in `git config` to get the behavior you want.

**Comment by technoweenie** [comment]
Since this issue is specifically about `clean` and `smudge` runs without STDIN, it will be closed when #264 lands.  Please do open another issue if you want to talk about a related issue (such as `.git/lfs/objects` cleanup) in more detail!
